components/connector-service/internal/monitoring/middleware/duration_middleware_test.go

```diff
@@ -0,0 +1,63 @@
+package middleware
+
+import (
+	"bytes"
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/kyma/components/connector-service/internal/monitoring/collector/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestDurationMiddleware_Handle(t *testing.T) {
+
+	t.Run("should observe status", func(t *testing.T) {
+		// given
+		path := "/v1/remoteenvironments/ec-default/tokens"
+
+		collector := &mocks.Collector{}
+		collector.On("AddObservation", mock.AnythingOfType("float64"), path, http.MethodPost).Return()
+
+		codeMiddleware := NewDurationMiddleware(collector)
+
+		router := mux.NewRouter()
+		router.Use(codeMiddleware.Handle)
+		router.Handle(path, getTestHandler())
+
+		testServer := httptest.NewServer(router)
+		defer testServer.Close()
+
+		var fullUrl bytes.Buffer
```

Comment:
consider using fmt.Sprintf instead of bytes.Buffer

Reply:
Done